R/complex_conditional_linter.R

```diff
@@ -0,0 +1,105 @@
+#' Complex Conditional Expressions Linter
+#'
+#' Detects complex conditional expressions and suggests extracting
+#' them into Boolean functions or variables for improved readability and reusability.
+#'
+#' For example, if you have a conditional expression with more than two logical operands,
+#'
+#' ```
+#' if (looks_like_a_duck(x) &&
+#'     swims_like_a_duck(x) &&
+#'     quacks_like_a_duck(x)) {
+#'     ...
+#' }
+#' ````
+#'
+#' to improve its readability and reusability, you can extract the conditional expression.
+#'
+#' You can either extract it into a Boolean function:
+#'
+#' ```
+#' is_duck <- function(x) {
+#'   looks_like_a_duck(x) &&
+#'     swims_like_a_duck(x) &&
+#'     quacks_like_a_duck(x)
+#' }
+#'
+#' if (is_duck(x)) {
+#'   ...
+#' }
+#' ```
+#'
+#' or into a Boolean variable:
+#'
+#' ```
+#' is_duck <- looks_like_a_duck(x) &&
+#'    swims_like_a_duck(x) &&
+#'    quacks_like_a_duck(x)
+#'
+#' if (is_duck) {
+#'  ...
+#' }
+#' ```
+#'
+#' In addition to improving code readability, extracting complex conditional expressions
+#' has the added benefit of introducing a reusable abstraction.
+#'
+#' @param threshold Integer. The maximum number of logical operators (`&&` or `||`)
+#'   allowed in a conditional expression. The default is `1L`, meaning any conditional expression
+#'   with more than one logical operator will be flagged.
```

Comment:
@MichaelChirico, @AshesITR WDYT? 

I can't make up my mind if the threshold should represent the number of **operators** or **operands**. 
I have chosen the former because I _feel_ that it's an easier and user-friendly way to detect the complexity of the conditional expression, but I am not wedded to the idea.